## Which jobs grew in pay after adjusting for inflation. 

In [1]:
import sqlite3
import os
import pandas as pd

In [2]:
occupation_file_name = os.path.join('Occupation_Data.csv')

In [3]:
occupation = pd.read_csv(occupation_file_name, index_col=None)

In [4]:
occupation.head()

OCC_CODE                            OCC_TITLE OCC_GROUP   TOT_EMP  EMP_PRSE  \
0  00-0000                      All Occupations     total  652100.0       1.3   
1  11-0000               Management Occupations     major   31180.0       2.6   
2  11-1011                     Chief Executives  detailed    1210.0       6.1   
3  11-1021      General and Operations Managers  detailed   10360.0       4.4   
4  11-2011  Advertising and Promotions Managers  detailed      60.0      16.8   

   H_MEAN    A_MEAN  MEAN_PRSE  H_PCT10  H_PCT25  H_MEDIAN  H_PCT75  H_PCT90  \
0   21.99   45740.0        1.5     9.18    11.90     17.47    26.74    38.71   
1   49.67  103320.0        1.5    19.90    28.60     42.37    61.46    87.13   
2   73.53  152940.0        2.8    28.36    41.72     61.92    95.40      NaN   
3   48.99  101900.0        2.1    17.27    24.89     38.13    61.21    96.22   
4   53.04  110320.0       10.4    18.64    26.74     44.61    78.86    98.10   

   A_PCT10  A_PCT25  A_MEDIAN   A_PCT75   A_PCT90  YEAR  
0  19100.0  24760.0   36350.0   55610.0   80510.0  2018  
1  41400.0  59490.0   88120.0  127850.0  181230.0  2018  
2  58980.0  86770.0  128800.0  198420.0       NaN  2018  
3  35920.0  51770.0   79310.0  127320.0  200130.0  2018  
4  38770.0  55630.0   92780.0  164030.0  204050.0  2018

In economics the term real dollars is the value of currency after being adjusted for inflation. I will turn everything into 2018 dollars and to compute the difference I used an inflation calculator which can be found in the following website 
https://www.usinflationcalculator.com/

Adjusted for inflation, 1.00 in 2003 is equal to 1.36 in 2018, with a 36.5% cumulative rate of inflation.

Adjusted for inflation, 1.00 in 2008 is equal to 1.17 in 2018, with a 16.6% cumulative rate of inflation.

Adjusted for inflation, 1.00 in 2013 is equal to 1.08 in 2018, with a 7.8% cumulative rate of inflation.

The following sections will need to be adjusted based on their year. 
H_PCT10, H_PCT25, H_MEDIAN, H_CT75, H_PCT90, A_PCT10, A_PCT25, A_MEDIAN, A_PCT75, A_PCT90

In [5]:
year2018 = occupation['YEAR'] == 2018
year2013 = occupation['YEAR'] == 2013
year2008 = occupation['YEAR'] == 2008
year2003 = occupation['YEAR'] == 2003

## Converting to 2018 dollars

In [6]:
occupation.columns[5:18]

for x in occupation.columns[5:18]:
    occupation.loc[(occupation.YEAR == 2013), x] *=1.08
    occupation.loc[(occupation.YEAR == 2008), x] *=1.17
    occupation.loc[(occupation.YEAR == 2003), x] *=1.36


## Checking to see how many occupations existed across all four sample years

In [7]:
occ_count=occupation.groupby('OCC_CODE')
occ_count_all = occ_count.size()
occ_count_all.value_counts()

4    384
2    139
3    128
1     80
5      5
dtype: int64

## Found an error in the data I need to find out where it is and remove it. (Skip to next section if you wish to skip this)
It is odd that there is a 5 in the options I need to figure out which year(s) have duplicate occupations.

In [8]:
dupe_year=occupation.groupby(['OCC_CODE', 'YEAR']).size() > 1
print(dupe_year.value_counts())
dupe_year_list=dupe_year[dupe_year == True]
dupe_year_list

False    2293
True        5
dtype: int64


OCC_CODE  YEAR
53-7061   2003    True
53-7062   2003    True
53-7063   2003    True
53-7064   2003    True
53-7081   2003    True
dtype: bool

2003 Seems to have produced all of the duplicates and here is the list of duplicate job codes 

I am going to have to remove the duplicate data from the dataset

In [9]:
dupes_year = occupation[occupation.duplicated(('OCC_CODE', 'YEAR'), keep=False)]

In [10]:
dupes_year.sort_values('OCC_CODE')

OCC_CODE                                          OCC_TITLE OCC_GROUP  \
2292  53-7061                 Cleaners of vehicles and equipment       NaN   
2298  53-7061                 Cleaners of vehicles and equipment       NaN   
2293  53-7062  Laborers and freight, stock, and material move...       NaN   
2299  53-7062  Laborers and freight, stock, and material move...       NaN   
2294  53-7063                     Machine feeders and offbearers       NaN   
2300  53-7063                     Machine feeders and offbearers       NaN   
2295  53-7064                        Packers and packagers, hand       NaN   
2301  53-7064                        Packers and packagers, hand       NaN   
2296  53-7081          Refuse and recyclable material collectors       NaN   
2302  53-7081          Refuse and recyclable material collectors       NaN   

      TOT_EMP  EMP_PRSE   H_MEAN   A_MEAN  MEAN_PRSE  H_PCT10  H_PCT25  \
2292   1260.0      15.7  13.1784  27404.0      6.664   8.1192   9.3432   
2298   1320.0      12.4  13.9264  28954.4     10.336   8.0784   9.2344   
2293  12700.0       5.6  15.3000  31824.0      6.120   9.5608  11.2200   
2299  12600.0       9.3  15.6672  32572.0      5.304   9.5336  11.4512   
2294   1650.0      29.6  17.0952  35550.4      5.848  11.0432  13.0016   
2300   1410.0      29.9  17.0680  35509.6      6.936  10.7848  12.6888   
2295   4970.0      11.3  11.9952  24956.0      2.312   8.2008   9.4248   
2301   4650.0       9.0  12.1448  25255.2      3.128   8.2280   9.3432   
2296    400.0      29.9  16.1976  33687.2      6.936  10.4992  12.8520   
2302    420.0      29.0  15.7488  32748.8      8.024  10.7032  12.2672   

      H_MEDIAN  H_PCT75  H_PCT90  A_PCT10  A_PCT25  A_MEDIAN  A_PCT75  \
2292   11.5736  15.3408  19.6656  16891.2  19434.4   24085.6  31905.6   
2298   11.5464  16.1160  24.1672  16796.0  19203.2   24017.6  33524.0   
2293   13.4912  16.4016  26.5336  19883.2  23337.6   28070.4  34122.4   
2299   14.0080  17.9520  26.1800  19842.4  23813.6   29131.2  37345.6   
2294   14.9736  20.4136  24.3304  22956.8  27036.8   31130.4  42445.6   
2300   14.8512  20.4272  24.2624  22426.4  26397.6   30885.6  42486.4   
2295   11.3832  13.9264  17.5168  17054.4  19611.2   23691.2  28968.0   
2301   11.3016  14.1984  18.2784  17095.2  19434.4   23514.4  29525.6   
2296   15.9120  18.8088  22.6032  21855.2  26724.0   33088.8  39127.2   
2302   14.5384  18.4824  22.1816  22249.6  25513.6   30232.8  38460.8   

      A_PCT90  YEAR  
2292  40908.8  2003  
2298  50252.0  2003  
2293  55188.8  2003  
2299  54454.4  2003  
2294  50605.6  2003  
2300  50456.0  2003  
2295  36420.8  2003  
2301  38025.6  2003  
2296  47001.6  2003  
2302  46131.2  2003

Much to my suprise the data is different. I suppose data gathering methods were not refined in 2003. To correct this I will merge the data and take the average. Once I get the averages I will add this to a new data base and remove all other duplicates. 

In [11]:
#Had to change the as_index to False so it would create number indexs rather than use the OOC_code
combine = dupes_year.groupby(['OCC_CODE', 'OCC_TITLE', 'YEAR'], as_index=False).mean()
combine = combine.copy()

In [12]:
combine['OCC_GROUP']=pd.np.NaN

In [13]:
new_occupation= occupation.append(combine, ignore_index=True, sort=False)

In [14]:
new_occupation.drop_duplicates(('OCC_CODE', 'YEAR'), keep='last', inplace=True)

In [15]:
occ_count=new_occupation.groupby('OCC_CODE')
occ_count_all = occ_count.size()
occ_count_4=occ_count.size() == 4
occ_count_3=occ_count.size() == 3
occ_count_2=occ_count.size() == 2
occ_count_1=occ_count.size() == 1

In [16]:
print(occ_count_all.value_counts())

4    389
2    139
3    128
1     80
dtype: int64


### Problem Solved
The problem has been solved. I know this becasue the 5 group no longer exists and the 4 group increased in size by five (the same number as the old 5 group). 
There are 736 unique occupations on this list of those occupations 389 occupations existed all four observed years, 128 existed for three observed years, 139 exitsed for two observed years, and 80 existed for only one observed year. 

In [17]:
#made dictionaries that can be used to show years
total_instance=new_occupation.groupby(['OCC_CODE'])
total_instance=total_instance.size()
dict4={}
dict3={}
dict2={}
dict1={}
for y, x in total_instance.items():
    if x == 4:
        dict4.update({y:x})
    elif x == 3:
        dict3.update({y:x})
    elif x == 2:
        dict2.update({y:x})
    else:
        dict1.update({y:x})   

In [18]:
check=(new_occupation.groupby(['OCC_CODE','YEAR'])).size()

In [19]:
occ_yeardict={}
for y,x in check.items():
    (a,b)=y
    if a in occ_yeardict:
        t=occ_yeardict[a]
        t = t + (b,)
        occ_yeardict.update({a:t})
    else:
        occ_yeardict.update({a:(b,)})
    

In [20]:
occ_yeardict

{'00-0000': (2003, 2008, 2013, 2018),
 '11-0000': (2003, 2008, 2013, 2018),
 '11-1011': (2003, 2008, 2013, 2018),
 '11-1021': (2003, 2008, 2013, 2018),
 '11-1031': (2003, 2008, 2013),
 '11-2011': (2003, 2008, 2013, 2018),
 '11-2021': (2003, 2008, 2013, 2018),
 '11-2022': (2003, 2008, 2013, 2018),
 '11-2031': (2003, 2008, 2013, 2018),
 '11-3011': (2003, 2008, 2013, 2018),
 '11-3021': (2003, 2008, 2013, 2018),
 '11-3031': (2008, 2013, 2018),
 '11-3040': (2003,),
 '11-3041': (2008,),
 '11-3042': (2008,),
 '11-3049': (2008,),
 '11-3051': (2003, 2008, 2013, 2018),
 '11-3061': (2003, 2008, 2013, 2018),
 '11-3071': (2003, 2008, 2013, 2018),
 '11-3111': (2013, 2018),
 '11-3121': (2013, 2018),
 '11-3131': (2013, 2018),
 '11-9021': (2003, 2008, 2013, 2018),
 '11-9031': (2003, 2008, 2013, 2018),
 '11-9032': (2003, 2008, 2013, 2018),
 '11-9033': (2008, 2013, 2018),
 '11-9039': (2008, 2013, 2018),
 '11-9041': (2003, 2008, 2013, 2018),
 '11-9051': (2003, 2008, 2013, 2018),
 '11-9061': (2003, 2008, 2

In [21]:
dictgap={}
for x, y in occ_yeardict.items():
    z=pd.np.diff(y)
    if (len(z)>0) and ((10 in z)or(15 in z)):
        if z[0]>5:
            dictgap.update({x:(y[0:2],z[0])})
        elif z[1]>5:
            dictgap.update({x:(y[1:],z[1])})
            
dictgap


{'13-1011': ((2003, 2018), 15),
 '19-1012': ((2008, 2018), 10),
 '19-1029': ((2008, 2018), 10),
 '19-2021': ((2008, 2018), 10),
 '19-2042': ((2003, 2013), 10),
 '21-1029': ((2008, 2018), 10),
 '23-1023': ((2003, 2013), 10),
 '25-1054': ((2003, 2013), 10),
 '25-2032': ((2008, 2018), 10),
 '25-9021': ((2008, 2018), 10),
 '27-1013': ((2008, 2018), 10),
 '27-1014': ((2003, 2018), 15),
 '27-1027': ((2003, 2013), 10),
 '27-2021': ((2008, 2018), 10),
 '27-4014': ((2003, 2013), 10),
 '27-4031': ((2008, 2018), 10),
 '29-2051': ((2008, 2018), 10),
 '29-2053': ((2003, 2018), 15),
 '39-5011': ((2003, 2018), 15),
 '39-6012': ((2003, 2013), 10),
 '41-9021': ((2003, 2018), 15),
 '41-9091': ((2003, 2013), 10),
 '45-2011': ((2008, 2018), 10),
 '45-2041': ((2008, 2018), 10),
 '47-2011': ((2008, 2018), 10),
 '47-2082': ((2008, 2018), 10),
 '47-4021': ((2003, 2013), 10),
 '49-2021': ((2003, 2013), 10),
 '49-9031': ((2003, 2013), 10),
 '49-9063': ((2008, 2018), 10),
 '51-4012': ((2008, 2018), 10),
 '51-405

In [22]:
len(dictgap)

43

In [23]:
new_occupation.sort_values(['YEAR', 'OCC_CODE'], ascending=[1, 1], inplace=True)
new_occupation

OCC_CODE                                          OCC_TITLE OCC_GROUP  \
1797  00-0000                                    All Occupations       NaN   
1798  11-0000                             Management occupations     major   
1799  11-1011                                   Chief executives       NaN   
1800  11-1021                    General and operations managers       NaN   
1801  11-1031                                        Legislators       NaN   
1802  11-2011                Advertising and promotions managers       NaN   
1803  11-2021                                 Marketing managers       NaN   
1804  11-2022                                     Sales managers       NaN   
1805  11-2031                          Public relations managers       NaN   
1806  11-3011                   Administrative services managers       NaN   
1807  11-3021          Computer and information systems managers       NaN   
1808  11-3040                           Human resources managers       NaN   
1809  11-3051                     Industrial production managers       NaN   
1810  11-3061                                Purchasing managers       NaN   
1811  11-3071  Transportation, storage, and distribution mana...       NaN   
1812  11-9021                              Construction managers       NaN   
1813  11-9031  Education administrators, preschool and child ...       NaN   
1814  11-9032  Education administrators, elementary and secon...       NaN   
1815  11-9041                               Engineering managers       NaN   
1816  11-9051                              Food service managers       NaN   
1817  11-9061                                  Funeral directors       NaN   
1818  11-9081                                   Lodging managers       NaN   
1819  11-9111               Medical and health services managers       NaN   
1820  11-9121                          Natural sciences managers       NaN   
1821  11-9131               Postmasters and mail superintendents       NaN   
1822  11-9141  Property, real estate, and community associati...       NaN   
1823  11-9151              Social and community service managers       NaN   
1824  13-0000      Business and financial operations occupations     major   
1825  13-1011  Agents and business managers of artists, perfo...       NaN   
1826  13-1021        Purchasing agents and buyers, farm products       NaN   
...       ...                                                ...       ...   
550   51-9194                              Etchers and Engravers  detailed   
551   51-9195  Molders, Shapers, and Casters, Except Metal an...  detailed   
552   51-9196  Paper Goods Machine Setters, Operators, and Te...  detailed   
553   51-9197                                      Tire Builders  detailed   
554   51-9198                        Helpers--Production Workers  detailed   
555   51-9199                      Production Workers, All Other  detailed   
556   53-0000     Transportation and Material Moving Occupations     major   
557   53-1011                Aircraft Cargo Handling Supervisors  detailed   
558   53-1048  First-Line Supervisors of Transportation and M...  detailed   
559   53-2012                                  Commercial Pilots  detailed   
560   53-3021                 Bus Drivers, Transit and Intercity  detailed   
561   53-3022              Bus Drivers, School or Special Client  detailed   
562   53-3031                               Driver/Sales Workers  detailed   
563   53-3032            Heavy and Tractor-Trailer Truck Drivers  detailed   
564   53-3033           Light Truck or Delivery Services Drivers  detailed   
565   53-3041                        Taxi Drivers and Chauffeurs  detailed   
566   53-3099                 Motor Vehicle Operators, All Other  detailed   
567   53-6021                             Parking Lot Attendants  detailed   
568   53-6031       Automotive and Watercraft Service Attendants  detailed   
569   53-6051              

In [31]:
new_occupation['H_CHANGE'] = new_occupation.groupby('OCC_CODE').H_MEAN.pct_change()
new_occupation

OCC_CODE                                          OCC_TITLE OCC_GROUP  \
1797  00-0000                                    All Occupations       NaN   
1798  11-0000                             Management occupations     major   
1799  11-1011                                   Chief executives       NaN   
1800  11-1021                    General and operations managers       NaN   
1801  11-1031                                        Legislators       NaN   
1802  11-2011                Advertising and promotions managers       NaN   
1803  11-2021                                 Marketing managers       NaN   
1804  11-2022                                     Sales managers       NaN   
1805  11-2031                          Public relations managers       NaN   
1806  11-3011                   Administrative services managers       NaN   
1807  11-3021          Computer and information systems managers       NaN   
1808  11-3040                           Human resources managers       NaN   
1809  11-3051                     Industrial production managers       NaN   
1810  11-3061                                Purchasing managers       NaN   
1811  11-3071  Transportation, storage, and distribution mana...       NaN   
1812  11-9021                              Construction managers       NaN   
1813  11-9031  Education administrators, preschool and child ...       NaN   
1814  11-9032  Education administrators, elementary and secon...       NaN   
1815  11-9041                               Engineering managers       NaN   
1816  11-9051                              Food service managers       NaN   
1817  11-9061                                  Funeral directors       NaN   
1818  11-9081                                   Lodging managers       NaN   
1819  11-9111               Medical and health services managers       NaN   
1820  11-9121                          Natural sciences managers       NaN   
1821  11-9131               Postmasters and mail superintendents       NaN   
1822  11-9141  Property, real estate, and community associati...       NaN   
1823  11-9151              Social and community service managers       NaN   
1824  13-0000      Business and financial operations occupations     major   
1825  13-1011  Agents and business managers of artists, perfo...       NaN   
1826  13-1021        Purchasing agents and buyers, farm products       NaN   
...       ...                                                ...       ...   
550   51-9194                              Etchers and Engravers  detailed   
551   51-9195  Molders, Shapers, and Casters, Except Metal an...  detailed   
552   51-9196  Paper Goods Machine Setters, Operators, and Te...  detailed   
553   51-9197                                      Tire Builders  detailed   
554   51-9198                        Helpers--Production Workers  detailed   
555   51-9199                      Production Workers, All Other  detailed   
556   53-0000     Transportation and Material Moving Occupations     major   
557   53-1011                Aircraft Cargo Handling Supervisors  detailed   
558   53-1048  First-Line Supervisors of Transportation and M...  detailed   
559   53-2012                                  Commercial Pilots  detailed   
560   53-3021                 Bus Drivers, Transit and Intercity  detailed   
561   53-3022              Bus Drivers, School or Special Client  detailed   
562   53-3031                               Driver/Sales Workers  detailed   
563   53-3032            Heavy and Tractor-Trailer Truck Drivers  detailed   
564   53-3033           Light Truck or Delivery Services Drivers  detailed   
565   53-3041                        Taxi Drivers and Chauffeurs  detailed   
566   53-3099                 Motor Vehicle Operators, All Other  detailed   
567   53-6021                             Parking Lot Attendants  detailed   
568   53-6031       Automotive and Watercraft Service Attendants  detailed   
569   53-6051              

In [39]:
#new_occupation['YEAR'] = pd.to_datetime(new_occupation['YEAR'], format = "%Y")

In [40]:
new_occupation.to_sql("occ_table",sqlite3.connect('occ.db'), if_exists ="replace")

In [41]:
con = sqlite3.connect("occ.db")

In [48]:
pd.read_sql_query("SELECT OCC_CODE,OCC_TITLE,AVG(H_CHANGE) as average FROM occ_table GROUP BY OCC_CODE ORDER BY average DESC", con)

OCC_CODE                                          OCC_TITLE   average
0    27-3011                    Radio and television announcers  1.131046
1    19-1023                 Zoologists and Wildlife Biologists  0.899938
2    19-1042         Medical Scientists, Except Epidemiologists  0.763306
3    21-1029                          Social workers, all other  0.521575
4    29-2091                        Orthotists and Prosthetists  0.419233
5    15-1134                                     Web Developers  0.412409
6    29-1121                                       Audiologists  0.402714
7    13-1011  Agents and business managers of artists, perfo...  0.396526
8    27-2042                              Musicians and singers  0.355757
9    13-1072  Compensation, benefits, and job analysis speci...  0.347175
10   51-4035  Milling and planing machine setters, operators...  0.325323
11   47-2161                       Plasterers and stucco masons  0.296543
12   51-8012                 Power distributors and dispatchers  0.283167
13   19-3039                           Psychologists, All Other  0.282993
14   19-2099                     Physical Scientists, All Other  0.280501
15   17-3024                     Electro-mechanical technicians  0.271004
16   51-3099                 Food Processing Workers, All Other  0.263949
17   41-9021                                Real estate brokers  0.257853
18   19-4061                 Social science research assistants  0.256483
19   19-3099   Social scientists and related workers, all other  0.255244
20   51-4051       Metal-refining furnace operators and tenders  0.234965
21   11-3131                  Training and Development Managers  0.232344
22   27-2012                            Producers and directors  0.227255
23   11-9061                                  Funeral directors  0.220125
24   53-5021       Captains, mates, and pilots of water vessels  0.212689
25   47-4021                  Elevator installers and repairers  0.208653
26   49-9098  Helpers--installation, maintenance, and repair...  0.190839
27   51-8013                              Power plant operators  0.183723
28   19-4021                             Biological Technicians  0.178034
29   15-1152               Computer Network Support Specialists  0.177271
..       ...                                                ...       ...
706  45-4022                        Logging Equipment Operators       NaN
707  45-4023                            Log Graders and Scalers       NaN
708  47-2042  Floor Layers, Except Carpet, Wood, and Hard Tiles       NaN
709  47-2043                        Floor sanders and finishers       NaN
710  47-2130                                 Insulation workers       NaN
711  47-3014  Helpers--Painters, Paperhangers, Plasterers, a...       NaN
712  49-2095  Electrical and Electronics Repairers, Powerhou...       NaN
713  49-3043                                 Rail Car Repairers       NaN
714  49-3091                                  Bicycle repairers       NaN
715  49-9011                          Mechanical Door Repairers       NaN
716  49-9064                                    Watch repairers       NaN
717  49-9095   Manufactured building and mobile home installers       NaN
718  51-2023             Electromechanical Equipment Assemblers       NaN
719  51-2028  Electrical, Electronic, and Electromechanical ...       NaN
720  51-2031                Engine and Other Machine Assemblers       NaN
721  51-2098  Assemblers and Fabricators, All Other, Includi...       NaN
722  51-3023                      Slaughterers and meat packers       NaN
723  51-6041             Shoe and Leather Workers and Repairers       NaN
724  51-6091  Extruding and forming machine setters, operato...       NaN
725  51-9151  Photographic Process Workers and Processing Ma...       NaN
726  51-9197                                      Tire Builders       NaN
727  53-1048  First-Line Supervisors of Transportation and M...       NaN
728  53-

In [47]:
pd.read_sql_query("SELECT * FROM occ_table WHERE OCC_CODE ='19-1023'", con)

index OCC_CODE                           OCC_TITLE OCC_GROUP  TOT_EMP  \
0    691  19-1023  Zoologists and Wildlife Biologists  detailed      NaN   
1    107  19-1023  Zoologists and Wildlife Biologists  detailed     60.0   

   EMP_PRSE   H_MEAN   A_MEAN  MEAN_PRSE  H_PCT10  ...  H_PCT75  H_PCT90  \
0       NaN  18.7164  38923.2      4.536   16.146  ...   19.062  19.7316   
1      47.0  35.5600  73960.0      5.400   21.510  ...   38.410  68.0700   

   A_PCT10  A_PCT25  A_MEDIAN  A_PCT75   A_PCT90  YEAR  H_CHANGE  H_Average  
0  33577.2  34981.2   37314.0  39646.8   41050.8  2013       NaN       None  
1  44750.0  52760.0   66820.0  79880.0  141590.0  2018  0.899938       None  

[2 rows x 22 columns]

In [29]:
occupation_2018 = pd.read_sql_query("SELECT * FROM occ_table WHERE YEAR=2018", con)
occupation_2013 = pd.read_sql_query("SELECT * FROM occ_table WHERE YEAR=2013", con)
occupation_2008 = pd.read_sql_query("SELECT * FROM occ_table WHERE YEAR=2008", con)
occupation_2003 = pd.read_sql_query("SELECT * FROM occ_table WHERE YEAR=2003", con)
type(occupation_2018)

pandas.core.frame.DataFrame

In [30]:
occ_yeardict

for x, y in occ_yeardict.items():
    years_check=len(y)
    if x not in dictgap:
        if years_check == 4:
            (y) = (y_1, y_2, y_3, y_4)
        elif years_check == 3:
        elif years_check == 2:
        else:

IndentationError: expected an indented block (<ipython-input-30-9726873ec765>, line 9)